File: Sources/RemDesignSystem/Brand/RemFaceMark.figma.swift
import Figma
import SwiftUI

// Code Connect for RemFaceMark — binds the Figma RemFaceMark component `362:7` (property `Mode` =
// idle / thinking) to the real Swift type (`RemFaceMark`, `RemFaceMark.Mode`). Co-located with the
// component so a rename here surfaces drift immediately, and `figma connect check` validates it
// against the node.
//
// EXCLUDED from the RemDesignSystem SPM target (see Package.swift), so the shipping library never
// links `github.com/figma/code-connect`. The `figma connect` CLI reads it directly. Publishing is
// plan-gated (Org/Enterprise + Dev/Full seat), so this stays dormant-but-ready — same as RemButton
// and ContainedIcon.
struct RemFaceMark_connection: FigmaConnect {
    let component = RemFaceMark.self
    let figmaNodeUrl =
        "https://www.figma.com/design/af4yDqCzp57jds9lkFiIaO/Rem-Design-System?node-id=362-7"

    @FigmaEnum("Mode", mapping: [
        "Idle": RemFaceMark.Mode.idle,
        "Thinking": RemFaceMark.Mode.thinking,
    ])
    var mode: RemFaceMark.Mode = .idle

    var body: some View {
        RemFaceMark(mode: mode, tint: DesignTokens.Color.labelPrimary, size: 96)
    }
}

File: compose/RemDesignSystem/brand/RemFaceMark.figma.kt
package com.rem.designsystem.brand

import androidx.compose.runtime.Composable
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

/**
 * Code Connect for `RemFaceMark` — the Compose twin of `RemFaceMark.figma.swift`. Binds the Figma
 * RemFaceMark component `362:7` (property `Mode` = idle / thinking) to the real Kotlin type, co-located
 * so a rename surfaces drift immediately (`figma connect check`).
 *
 * Like the Swift `.figma.swift` files, this is **excluded from the library's Gradle build** (the app
 * never links `com.figma.code.connect`; the `gatherDesignSystemSources` task drops `**/*.figma.kt`);
 * the `figma connect` CLI reads it directly. Dormant-but-ready — publishing is plan-gated (same as
 * RemButton / ContainedIcon / the SwiftUI Code Connect).
 *
 * ⚠️ VERIFY ON THE ANDROID RUNNER: confirm the exact `@FigmaProperty` enum-mapping signature against
 * the pinned `com.figma.code.connect` version. The SwiftUI `RemFaceMark.figma.swift` is the mature
 * reference for the mapping intent:  Mode: "Idle" → Idle, "Thinking" → Thinking.
 */
@FigmaConnect("https://www.figma.com/design/af4yDqCzp57jds9lkFiIaO/Rem-Design-System?node-id=362-7")
class RemFaceMarkDoc {
    @FigmaProperty(FigmaType.Enum, "Mode")
    val mode: RemFaceMarkMode = RemFaceMarkMode.Idle

    @Composable
    fun example() {
        RemFaceMark(mode = mode)
    }
}
